ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpServerSentEventsSession.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.okhttp
+
+import io.ktor.client.plugins.sse.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import okhttp3.*
+import okhttp3.sse.*
+import kotlin.coroutines.*
+
+internal class OkHttpServerSentEventsSession(
+    engine: OkHttpClient,
+    engineRequest: Request,
+    override val coroutineContext: CoroutineContext,
+) : ClientServerSentEventsSession, EventSourceListener() {
+    private val serverSentEventsSource = EventSources.createFactory(engine).newEventSource(engineRequest, this)
+
+    internal val originResponse: CompletableDeferred<Response> = CompletableDeferred()
+
+    private val _incoming = Channel<ServerSentEvent>()
+
+    override val incoming: ReceiveChannel<ServerSentEvent>
+        get() = _incoming
+
+    override fun onOpen(eventSource: EventSource, response: Response) {
+        super.onOpen(eventSource, response)
+        originResponse.complete(response)
+    }
+
+    override fun onEvent(eventSource: EventSource, id: String?, type: String?, data: String) {
+        super.onEvent(eventSource, id, type, data)
+        _incoming.trySendBlocking(ServerSentEvent(type, id, data))
```

Comment:
I think, that it's better to use `Channel(Channel.UNLIMITED)` and use just `trySend` here instead of blocking variant